superset/connectors/sqla/models.py

```diff
@@ -159,14 +159,25 @@ def datasource(self) -> RelationshipProperty:
         return self.table
 
     def get_time_filter(
-        self, start_dttm: DateTime, end_dttm: DateTime
+        self,
+        start_dttm: DateTime,
+        end_dttm: DateTime,
+        time_range_endpoints: Optional[
+            Tuple[utils.TimeRangeEndpoint, utils.TimeRangeEndpoint]
+        ],
     ) -> ColumnElement:
         col = self.get_sqla_col(label="__time")
         l = []  # noqa: E741
         if start_dttm:
             l.append(col >= text(self.dttm_sql_literal(start_dttm)))
         if end_dttm:
-            l.append(col <= text(self.dttm_sql_literal(end_dttm)))
+            if (
+                time_range_endpoints
+                and time_range_endpoints[1] == utils.TimeRangeEndpoint.EXCLUSIVE
+            ):
+                l.append(col < text(self.dttm_sql_literal(end_dttm)))
+            else:
+                l.append(col <= text(self.dttm_sql_literal(end_dttm)))
```

Comment:
Shouldn't the start endpoint apply to `start_dttm`, too?

Reply:
@villebro the start endpoint is currently correct (though SIP-15/SIP-15A needs to address aspects where `[start` actually behaves like `(start`). I guess one could argue for the filtering we really only care about the upper endpoint, however I have defined both so we can more accurately display how we believe the time range filter is being applied.
`